lib/algos/list/delimiterMaster.js

```diff
@@ -0,0 +1,67 @@
+'use strict'; // eslint-disable-line strict
+
+const Delimiter = require('./delimiter').Delimiter;
+const VSUtils = require('../../versioning/utils').VersioningUtils;
+
+/**
+ * Extended delimiter class for versioning.
+ */
+class DelimiterMaster extends Delimiter {
+    addContents(key, value) {
+        this.Contents.push({
+            key,
+            value: {
+                Size: value['content-length'],
+                ETag: value['content-md5'],
+                LastModified: value['last-modified'],
+                VersionId: VSUtils.getts(value),
+                Owner: {
+                    DisplayName: value['owner-display-name'],
+                    ID: value['owner-id'],
+                },
+                StorageClass: value['x-amz-storage-class'],
+                Initiated: value.initiated,
+                Initiator: value.initiator,
+                EventualStorageBucket: value.eventualStorageBucket,
+                partLocations: value.partLocations,
+                creationDate: value.creationDate,
+            },
+        });
+        this.NextMarker = key;
+        ++this.keys;
+    }
+
+    filter(obj) {
```

Comment:
This is overriding the function of the base class. Some comment for the next commit should not be troublesome anyway.